membership-attribute-service/app/parsers/Salesforce.scala

```diff
@@ -1,48 +1,73 @@
 package parsers
 
 import models.Attributes
-
-import scala.xml._
+import scala.xml.{Node, _}
 import scalaz.Scalaz._
-import scalaz.\/
+import scalaz.{-\/, \/, \/-}
 
 object Salesforce {
   type UserId = String
 
   sealed trait OutboundMessageChange
+
   case class MembershipUpdate(attributes: Attributes) extends OutboundMessageChange
+
   case class MembershipDeletion(userId: String) extends OutboundMessageChange
 
   sealed trait OutboundMessageParseError
+
   case class OrgIdMatchingError(organizationId: String) extends OutboundMessageParseError
+
   case class ParsingError(msg: String) extends OutboundMessageParseError
 
   /**
-   * @param payload The outbound message content coming from Salesforce
-   * @param organizationId  The id that qualifies the originating Salesforce account. This is to make sure that
-   *                        the service do not process outbound messages from SF environments created by duplicating
-   *                        Prod
-   * @return
-   */
-  def parseOutboundMessage(payload: NodeSeq, organizationId: String): OutboundMessageParseError \/ OutboundMessageChange = {
+    * @param payload        The outbound message content coming from Salesforce
+    * @param organizationId The id that qualifies the originating Salesforce account. This is to make sure that
+    *                       the service do not process outbound messages from SF environments created by duplicating
+    *                       Prod
+    * @return
+    */
+  def parseOutboundMessage(payload: NodeSeq, organizationId: String): OutboundMessageParseError \/ Seq[OutboundMessageChange] = {
     implicit class NodeSeqOps(ns: NodeSeq) {
       def getTag(tag: String): OutboundMessageParseError \/ Node =
         (ns \ tag).headOption \/> ParsingError(s"Error while parsing the outbound message: $tag not found.\n $payload")
 
       def getText(tag: String): OutboundMessageParseError \/ String = getTag(tag).map(_.text)
     }
 
-    for {
-      orgId <- (payload \\ "notifications").getText("OrganizationId")
-      _ <- if (orgId === organizationId) ().right else OrgIdMatchingError(orgId).left
-      obj <- (payload \\ "Notification").getTag("sObject")
-      id <- obj.getText("IdentityID__c")
-    } yield {
-      val tier = (obj \ "Membership_Tier__c").map(_.text).headOption
-      tier.fold[OutboundMessageChange](MembershipDeletion(id)) { t =>
-        val num = (obj \ "Membership_Number__c").headOption.map(_.text)
-        MembershipUpdate(Attributes(id, t, num))
+    def orgIdValidation: OutboundMessageParseError \/ Boolean = {
```

Comment:
I'd probably called it orgIdValid (although it is the validation code, the function is evaluating to whether it's valid)  you can prefix with "is" although I heard it's more java style, but  I still do it sometimes